torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -42,13 +42,10 @@ struct TORCH_API AccumulateGrad : public Node {
 
   variable_list apply(variable_list&& grads) override;
 
-  static at::Tensor callHooks(const Variable& variable, at::Tensor new_grad) {
-    for (auto& hook : impl::hooks(variable)) {
-      new_grad = (*hook)({new_grad})[0];
-    }
-    return new_grad;
+  std::vector<std::unique_ptr<FunctionPreHook>>& tensor_pre_hooks() noexcept
```

Comment:
Shouldn't we have the same for retain_grad hook? Or is it assumed that leaf Tensors can't have retain_grad hooks?

Reply:
Yes, I think it is ok to assume that.